File: Project15/backend/src/models/Product.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/database');

const Product = sequelize.define('Product', {
  id: {
    type: DataTypes.INTEGER,
    primaryKey: true,
    autoIncrement: true
  },
  name: {
    type: DataTypes.STRING(100),
    allowNull: false,
    comment: '货品名称'
  },
  category: {
    type: DataTypes.STRING(50),
    comment: '分类'
  },
  spec: {
    type: DataTypes.STRING(100),
    comment: '规格'
  },
  unit: {
    type: DataTypes.STRING(20),
    comment: '单位'
  },
  price: {
    type: DataTypes.DECIMAL(10, 2),
    comment: '批发价'
  },
  supplierId: {
    type: DataTypes.INTEGER,
    comment: '供货商ID',
    field: 'supplier_id'
  },
  remark: {
    type: DataTypes.TEXT,
    comment: '备注'
  }
}, {
  tableName: 'products',
  timestamps: true,
  createdAt: 'created_at',
  updatedAt: 'updated_at'
});

module.exports = Product;

File: Project40/models/product.go
package models

import (
	"time"
)

type Product struct {
	ID            uint      `json:"id" gorm:"primaryKey"`
	Name          string    `json:"name" gorm:"not null;size:255"`
	Category      string    `json:"category" gorm:"size:100"`
	Brand         string    `json:"brand" gorm:"size:100"`
	Spec          string    `json:"spec" gorm:"size:255"`
	PurchasePrice float64   `json:"purchase_price" gorm:"type:decimal(10,2);not null"`
	SalePrice     float64   `json:"sale_price" gorm:"type:decimal(10,2);not null"`
	Stock         int       `json:"stock" gorm:"not null;default:0"`
	MinStock      int       `json:"min_stock" gorm:"not null;default:5"`
	Unit          string    `json:"unit" gorm:"size:20"`
	Remark        string    `json:"remark" gorm:"type:text"`
	CreatedAt     time.Time `json:"created_at"`
	UpdatedAt     time.Time `json:"updated_at"`
}

func (p *Product) IsLowStock() bool {
	return p.Stock <= p.MinStock
}
